File: intellij-lsp/src/com/github/gtache/lsp/contributors/rename/LSPRenameHelper.scala
package com.github.gtache.lsp.contributors.rename

import com.github.gtache.lsp.editor.EditorEventManager
import com.intellij.openapi.diagnostic.Logger
import com.intellij.openapi.editor.Editor
import com.intellij.openapi.ui.{Messages, NonEmptyInputValidator}

object LSPRenameHelper {
  private val LOG: Logger = Logger.getInstance(LSPRenameHelper.getClass)

  def rename(editor: Editor): Unit = {
    EditorEventManager.forEditor(editor) match {
      case Some(manager) =>
        if (manager.canRename()) {
          val renameTo = Messages.showInputDialog(editor.getProject, "Enter new name: ", "Rename", Messages.getQuestionIcon, "", new NonEmptyInputValidator())
          if (renameTo != null && renameTo != "") manager.rename(renameTo)
        }
      case None =>
    }
  }
}

File: intellij-lsp/src/main/kotlin/com/github/gtache/lsp/contributors/rename/LSPRenameHelper.kt
package com.github.gtache.lsp.contributors.rename

import com.github.gtache.lsp.editor.services.application.EditorApplicationService
import com.intellij.openapi.components.service
import com.intellij.openapi.editor.Editor
import com.intellij.openapi.ui.Messages
import com.intellij.openapi.ui.NonEmptyInputValidator

/**
 * LSP application rename helper
 */
object LSPRenameHelper {

    /**
     * Renames the currently selected element in the given editor
     */
    fun rename(editor: Editor): Unit {
        val manager = service<EditorApplicationService>().managerForEditor(editor)
        if (manager != null) {
            if (manager.canRename()) {
                val renameTo = Messages.showInputDialog(editor.project, "Enter name: ", "Rename", Messages.getQuestionIcon(), "", NonEmptyInputValidator())
                if (renameTo != null && renameTo != "") manager.rename(renameTo)
            }
        }
    }
}